### PR TITLE
Add CREATE_GET_STATIC_MEMBER_VARIABLE_OR_DEFAULT

### PR DESCRIPTION
**Pull request by wthrowe**
This has come up a few times in discussion recently, and I found myself wanting it as well.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Review by kidder — DISMISSED** [review]

File: src/Utilities/TypeTraits/CreateGetStaticMemberVariableOrDefault.hpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * \note This macro requires the type traits created by
+ * `CREATE_HAS_STATIC_MEMBER_VARIABLE` and
+ * `CREATE_HAS_STATIC_MEMBER_VARIABLE_V` but does not create them --
+ * both of those macros should be called before this macro to generate
+ * the needed type traits.
+ */
+#define CREATE_GET_STATIC_MEMBER_VARIABLE_OR_DEFAULT(ALIAS_NAME)          \
+  template <typename CheckingType, auto Default,                          \
+            bool present = has_##ALIAS_NAME##_v<CheckingType>>            \
```
Comment (nilsvu) [inline src/Utilities/TypeTraits/CreateGetStaticMemberVariableOrDefault.hpp:26]: Do you actually need `has_ALIAS_NAME_v`? I think you can replace this line with `typename = std::void_t<>` and in the specialization below replace the bool with `std::void_t<CheckingType::ALIAS_NAME>`, or am I missing something? Edit: Probably what I'm missing is, that my suggestion only works for type aliases?

File: src/Utilities/TypeTraits/CreateGetStaticMemberVariableOrDefault.hpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+#pragma once
+
+#include <type_traits>
+
+/*!
+ * \ingroup TypeTraitsGroup
+ * \brief Generate a metafunction that will retrieve the specified
+ * `static constexpr` member variable, or, if not present, a default.
+ *
```
Comment (nilsvu) [inline src/Utilities/TypeTraits/CreateGetStaticMemberVariableOrDefault.hpp:12]: I'm a bit worried that this helper makes it easier to create hidden defaults. How about adding a comment, something like this: `\warning Please use defaults responsibly. In many cases it is better to require the static member variable is present so the functionality isn't hidden. Also make sure to document the compile-time interface, e.g. with a \ref protocols "protocol".`

File: src/Utilities/TypeTraits/CreateGetStaticMemberVariableOrDefault.hpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <type_traits>
+
+/*!
+ * \ingroup TypeTraitsGroup
+ * \brief Generate a metafunction that will retrieve the specified
+ * `static constexpr` member variable, or, if not present, a default.
```
Comment (nilsvu) [inline src/Utilities/TypeTraits/CreateGetStaticMemberVariableOrDefault.hpp:11]: Would it be reasonably straight-forward to make this macro work for type aliases as well?

**Review by nilsvu — CHANGES_REQUESTED** [review]

**Comment by wthrowe** [comment]
@nilsleiffischer Take a look at `src/Utilities/TypeTraits/CreateGetTypeAliasOrDefault.hpp` on develop and let me know if you still have any questions/concerns.  I went for consistency here, rather than what I would have designed in a vacuum.

**Comment by nilsvu** [comment]
@wthrowe Ah I wasn't aware of that. That resolves my third comment. I think the other two comments are still valid, but please treat them as optional.

File: src/Utilities/TypeTraits/CreateGetStaticMemberVariableOrDefault.hpp (hunk 1 above)
Comment (wthrowe) [reply src/Utilities/TypeTraits/CreateGetStaticMemberVariableOrDefault.hpp:26]: If I were designing the interface, I would have done it that way, but I'm not going to redo existing interfaces in this PR.

(You can use decltype to turn the value into a type to SFINAE with.)

File: src/Utilities/TypeTraits/CreateGetStaticMemberVariableOrDefault.hpp (hunk 2 above)
Comment (wthrowe) [reply src/Utilities/TypeTraits/CreateGetStaticMemberVariableOrDefault.hpp:12]: I'll do this and add it to the type alias version as well.

**Review by nilsvu — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**